src/client/datascience/notebook/notebookEditorProvider.ts

```diff
@@ -131,28 +132,9 @@ export class NotebookEditorProvider implements INotebookEditorProvider {
     }
     @captureTelemetry(Telemetry.CreateNewNotebook, undefined, false)
     public async createNew(options?: { contents?: string; defaultCellLanguage: string }): Promise<INotebookEditor> {
-        const nbJson: nbformat.INotebookContent = {
-            cells: [],
-            metadata: { orig_nbformat: defaultNotebookFormat.major },
-            nbformat: defaultNotebookFormat.major,
-            nbformat_minor: defaultNotebookFormat.minor
-        };
-        if (options?.contents) {
-            Object.assign(nbJson, JSON.parse(options.contents));
-        }
-        if (options?.defaultCellLanguage) {
-            if (!nbJson.metadata) {
-                nbJson.metadata = nbJson.metadata || { orig_nbformat: 4 };
-            }
-            if (nbJson.metadata.language_info) {
```

Comment:
Do you need to keep this metadata?

Reply:
Yes, we need that, that's how we pick the preferred kernel (we indirect the metadata)

Reply:
I think we need to restore that code